index.d.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: strong-error-handler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// Type definitions for strong-error-handler 3.x
+// Project: https://github.com/strongloop/strong-error-handler
+// Definitions by: Kyusung Shim <https://github.com/shimks>
+// TypeScript Version: 2.9
+
+import * as Express from 'express';
+
+export = errorHandlerFactory;
+
+/**
+ * Creates a middleware function for error-handling
+ * @param options Options for error handler settings
+ */
+declare function errorHandlerFactory(
+  options?: errorHandlerFactory.ErrorHandlerOptions
+): errorHandlerFactory.StrongErrorHandler;
+
+declare namespace errorHandlerFactory {
+
+  /**
+   * Error-handling middleware function. Includes server-side logging
+   */
+  type StrongErrorHandler = (
```

Comment:
Can we move types to the top level please, so that users don't have to prefix their usage with `errorHandlerFactory` namespace?

I am referring to the following types:

 - `StrongErrorHandler`
 - `ErrorWriterOptions`
 - `ErrorHandlerOptions`

Of course, `writeErrorToResponse` must stay inside `errorHandlerFactory` namespace, because that's how the JavaScript code is structured.

```ts
declare interface ErrorWriterOptions {
  // ...
}

declare interface ErrorHandlerOptions {
  // ...
}

declare type StrongErrorHandler =
  // ...

declare function errorHandlerFactory
  // ...

declare namespace errorHandlerFactory {
  function writeErrorToResponse
    // ...
}
```

Reply:
If I understand correctly, moving the types outside of the namespace does not allow us to export the types since `module.exports` is only the handler function itself.

As long as the error handler function isn't used from the module, the users shoudn't have to use the namespace if that was what you were concerned about. See here https://github.com/strongloop/loopback-next/blob/4e180b286bf6a353067995795a7770522332588a/packages/rest/src/providers/reject.provider.ts#L10

Reply:
> If I understand correctly, moving the types outside of the namespace does not allow us to export the types since module.exports is only the handler function itself.

I see what you mean now! I was not aware of this limitation, thank you for explaining it to me.